macros/src/main/scala/pureconfig/Derivation.scala

```diff
@@ -81,15 +96,20 @@ class DerivationMacros(val c: whitebox.Context) extends LazyContextParser with M
   // found error with a basic message. If an implicit is found, it still needs to search the tree found in order to
   // check if some inner derivations materialized a `Derivation.failed`. If that's the case, it collects those failures
   // and prints a nice message.
-  private[this] def materializeRootDerivation(typ: Type): Tree = {
+  private[this] def materializeRootDerivation(typ: Type, isMaterializationExplicitCall: Boolean): Tree = {
```

Comment:
I think the code would become more elegant if you received here a `failImplicitSearch: String => Nothing` and have the `materializeDerivation` pass the correct strategy, instead of having implicit-vs-explicit logic on both sides.

Reply:
I agree that it becomes more elegant! I've refactored this in 92cc149.